### PR TITLE
Add pre-commit

### LINKED ISSUE #272: Add pre-commit
It is quite tedious to run `black` and `isort` formatter every time before adding a commit. With `pre-commit` this can be automated.

### PR DESCRIPTION
**Pull request by geek-yang**
Use pre-commit to automatically check and format code with `black` and `isort`.

Note that all the other checks which require manual corrections are not included.

This PR closes #272.

### CODE REVIEW

File: docs/develop.md [hunk 1]
```diff
@@ -162,6 +162,9 @@ git diff
 git add -u
 git add docs/develop.md
 
+# Run `pre-commit` before committing your changes
```
Comment (Peter9192) [inline docs/develop.md:165]: Perhaps we can extend this a little bit.

> To help you make clean commits, we have configured this repository with pre-commit. This is completely optional, but we recommend you give it a try. You can run pre-commit once by typing `pre-commit run` before committing some changes. If you like this, and you want to do it automatically with each commit, you can type `pre-commit install`.

**Review by Peter9192 — APPROVED** [review]
Like it. Left a suggestion to document it a bit better

**Comment by Peter9192** [comment]
@Imme1992 @ruudvdent @chrisweij we didn't discuss this last week but I think it's a nice addition. For more info on pre-commit, see https://pre-commit.com/

Thanks @geek-yang!

**Comment by ruudvdent** [comment]
cool

**Pull request merged**